### DISCUSSION TITLE
Buffalo WXR-11000XE12, anyone using this router?

### DISCUSSION THREAD

**Discussion by titanmaru** [fakemanhk/openwrt-jp-ipoe#26] [Q&A]
Hello, I'm using the said router with Docomo 10G. Lately I got randomly disconnected very often like once every 5-10 minutes and then got reconnected which is very annoying. 

So I'm looking to flash my router to OpenWRT. But seems like it is not yet supported. 

Anyone know anything about this router availability of firmware? Can I use WXR-6000AX12 firmware? I saw that listed and the difference I see between the 2 is only Wifi 6E. 

Any help or thoughts would be greatly appreciated. Thank you

**Comment by LIII-XXII** [comment]
based on experience with the 6000AX12:

you can (carefully) open the router, there should be a soldered header for TX/RX/GND of the UART. You can check the pictures here: https://openwrt.org/toh/buffalo/wxr-5950ax12
I expect the layout of the board to be very similar.

What you can do is:

- connect to the board UART
- power on the device with the UART attached
- compare the boot log with the OEM boot log from the wiki page

Since the wi-fi specs on the WXR-11000 are different, it is probably going to be another wireless IC, but it might be supported by the same driver. In case it is not, but the rest of the board is the same, you might be able to run the WXR-6000 openwrt build and install the correct driver via ethernet (or build a release with what you need).

Some notes with trying the WXR-5950 firmware on the WXR-6000:
- we could try the firmware without flashing by loading it at boot time from tftp, but the wifi did not work in this case because it needs some blobs from flash memory
- the wifi chip worked after flashing, although the 5GHz wifi had very low RX gain (clients could see the network but not connect), which was fixed by copying some data from flash to filesystem like so:
```
mv /lib/firmware/ath11k/IPQ8074/hw2.0/board-2.bin /lib/firmware/ath11k/IPQ8074/hw2.0/_IGNORE_board-2.bin
. /lib/functions/caldata.sh
FIRMWARE=ath11k/IPQ8074/hw2.0/board.bin caldata_extract "0:art" 0x1000 0x20000
reboot
```

see this thread for details https://forum.openwrt.org/t/qualcommax-ath11k-board-calibration-bdf-data-woes/182646

also see this git commit that adds support for the 5950: https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=f490295bf213c5e181d16cbe53c163c1c9983e45

**Comment by missing233** [comment]
I suspect that Buffalo's stock firmware is based on OpenWrt, because OpenWrt also has this issue.

**Reply by LIII-XXII** [reply]
that's because it most likely is. the stock firmware says so, you can see it in the boot log. for example for the [wxr-5950ax12](https://openwrt.org/toh/buffalo/wxr-5950ax12) (very similar model):

![image](https://github.com/fakemanhk/openwrt-jp-ipoe/assets/583664/03c0e139-2fdd-4726-82d8-5e858f9642e2)

**Comment by fakemanhk** [comment]
> I suspect that Buffalo's stock firmware is based on OpenWrt, because OpenWrt also has this issue.

Almost all home routers are kind of OpenWrt based (except a few ancient NEC one I've seen using BSD), they may or may not disclose (e.g. GL-INET, NanoPi, Asus advertise them using some kind of WRT variant), and if you look at some previous discussions here, someone grabbed an Elecom IPoE supported router's firmware and unpacked to find it also OpenWrt based.

However whether this can be officially supported by OpenWrt, depends on the whether bootloading process can be worked out or not, and the most important is SoC driver, no Linux kernel upstream, no chance on getting support.

**Comment by missing233** [comment]
> > I suspect that Buffalo's stock firmware is based on OpenWrt, because OpenWrt also has this issue.
> 
> Almost all home routers are kind of OpenWrt based (except a few ancient NEC one I've seen using BSD), they may or may not disclose (e.g. GL-INET, NanoPi, Asus advertise them using some kind of WRT variant), and if you look at some previous discussions here, someone grabbed an Elecom IPoE supported router's firmware and unpacked to find it also OpenWrt based.
> 
> However whether this can be officially supported by OpenWrt, depends on the whether bootloading process can be worked out or not, and the most important is SoC driver, no Linux kernel upstream, no chance on getting support.

I'm referring to the issue of frequent disconnections. Both my Xiaomi router and OpenWrt stock device experience this problem on flet's cross. Without identifying the cause, switching to OpenWrt won't solve it.

**Comment by powerichard** [comment]
你好，我也有同样的想法，想知道你成功了吗？
